### ISSUE TITLE
Missing `allowedPlaceholders` typing for manifest component definition

### ISSUE THREAD

**Issue by aweber1** [Sitecore/jss#120]
## Description

Developers can explicitly define which placeholders a component (rendering) is allowed to be added to via the manifest component definition. However, it's not very clear or obvious how to do so.

When defining components, you can add an `allowedPlaceholders` property to the component and the manifest generation will assign that property to the appropriate rendering object in the manifest. Also, deployment / import will recognize the property and assign "Allowed Controls" properly within Sitecore placeholder settings.

```
manifest.addComponent({
    name: 'My-Component',
    icon: SitecoreIcon.Compass,
    allowedPlaceholders: ['jss-is-awesome'],
  });
```

However, the TypeScript definition for the `ComponentDefinition` interface does not contain the `allowedPlaceholders` property. 

## Fix

Add `allowedPlaceholders` property to the `ComponentDefinition` interface in this file: https://github.com/Sitecore/jss/blob/dev/packages/sitecore-jss-manifest/src/generator/manifest.types.ts

Also, documentation should be updated, there are conflicting `component` definitions on this page: https://jss.sitecore.com/docs/techniques/working-disconnected/manifest-api

Perhaps also consider adding an example to the samples?

**Comment by kamsar** [comment]
Fixed in #134

**Comment by Afzal-Shaik** [comment]
![error](https://user-images.githubusercontent.com/82775470/124258679-4e712480-db4b-11eb-93de-e014628200b3.GIF)
